src/audio/module_adapter/iadk/iadk_module_adapter.cpp

```diff
@@ -80,7 +80,14 @@ uint32_t IadkModuleAdapter::IadkModuleAdapter_Process(struct sof_source **source
 			new (&output_stream_buffers[i]) intel_adsp::OutputStreamBuffer(osb_data);
 		}
 
-		ret = processing_module_.Process(input_stream_buffers, output_stream_buffers);
+		uint32_t iadk_ret =
```

Comment:
it is still be negative value for module process, still need use int.
may be should directly return processing_module_.Process(input_stream_buffers, output_stream_buffers);

Reply:
No. IADK modules are from reference FW. These are using uint32_t return values. 
See: https://github.com/thesofproject/sof/blob/2a0607c658c8b1c487ac63e8e2b5d3410766069a/src/include/sof/audio/module_adapter/iadk/processing_module_interface.h#L230C1-L243C9